src/Interpreters/Cache/FileCache.cpp

```diff
@@ -776,7 +782,17 @@ bool FileCache::tryReserve(
     ProfileEventTimeIncrement<Microseconds> watch(ProfileEvents::FilesystemCacheReserveMicroseconds);
 
     assertInitialized();
-    auto cache_lock = lockCache();
+    auto cache_lock = tryLockCache();
+    if (!cache_lock)
```

Comment:
This seems too radical to me, I would rather use `try_lock_for` with some reasonable timeout.